gtars/src/common/models/region_set.rs

```diff
@@ -25,14 +40,95 @@ impl TryFrom<&Path> for RegionSet {
     /// # Arguments:
     /// - value: path to bed file on disk.
     fn try_from(value: &Path) -> Result<Self> {
-        let regions = extract_regions_from_bed_file(value)?;
-        Ok(RegionSet { regions })
+        let path = value;
+        println!(
+            "Initiating regionSet object from: {}",
+            value.to_str().unwrap()
+        );
+
+        let mut new_regions: Vec<Region> = Vec::new();
+
+        let reader = match path.is_file() {
+            true => get_dynamic_reader(path).expect("!Can't read file"),
+            false => get_dynamic_reader_from_url(path).expect("!Can't read file or url!"),
```

Comment:
Doesnt expect panic? Can we just return the error? I might be wrong here

Reply:
I tried to do this, but unfortunately, they return different types, and rust is mad about it